File: test_samples/sample.go
package main

import (
	"fmt"
	"sync"
)

// User represents a user in the system
type User struct {
	ID    int
	Name  string
	Email string
}

// NewUser creates a new user with the given ID and name
func NewUser(id int, name string) *User {
	return &User{
		ID:    id,
		Name:  name,
		Email: "",
	}
}

// SetEmail sets the user's email address
func (u *User) SetEmail(email string) {
	u.Email = email
}

// Display returns a formatted display string
func (u *User) Display() string {
	return fmt.Sprintf("User %d: %s", u.ID, u.Name)
}

// UserManager manages a collection of users
type UserManager struct {
	mu    sync.Mutex
	users map[int]*User
}

// NewUserManager creates a new empty user manager
func NewUserManager() *UserManager {
	return &UserManager{
		users: make(map[int]*User),
	}
}

// AddUser adds a user to the manager
func (m *UserManager) AddUser(user *User) {
	m.mu.Lock()
	defer m.mu.Unlock()
	m.users[user.ID] = user
}

// GetUser retrieves a user by ID
func (m *UserManager) GetUser(id int) *User {
	m.mu.Lock()
	defer m.mu.Unlock()
	return m.users[id]
}

// ListAll returns all users
func (m *UserManager) ListAll() []*User {
	m.mu.Lock()
	defer m.mu.Unlock()
	
	users := make([]*User, 0, len(m.users))
	for _, user := range m.users {
		users = append(users, user)
	}
	return users
}

// RemoveUser removes a user by ID
func (m *UserManager) RemoveUser(id int) bool {
	m.mu.Lock()
	defer m.mu.Unlock()
	
	if _, exists := m.users[id]; exists {
		delete(m.users, id)
		return true
	}
	return false
}

// main is the entry point of the application
func main() {
	manager := NewUserManager()
	
	user1 := NewUser(1, "Alice")
	user2 := NewUser(2, "Bob")
	
	manager.AddUser(user1)
	manager.AddUser(user2)
	
	for _, user := range manager.ListAll() {
		fmt.Println(user.Display())
	}
}

File: test_samples/sample.c
/**
 * Sample C module demonstrating C parsing capabilities
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/**
 * User structure
 */
typedef struct {
    int id;
    char name[50];
    char email[100];
} User;

/**
 * Creates a new user with the given ID and name
 * @param id The user ID
 * @param name The user name
 * @return Pointer to new user or NULL on failure
 */
User* user_new(int id, const char* name) {
    User* user = (User*)malloc(sizeof(User));
    if (!user) return NULL;
    
    user->id = id;
    strncpy(user->name, name, sizeof(user->name) - 1);
    user->name[sizeof(user->name) - 1] = '\0';
    user->email[0] = '\0';
    
    return user;
}

/**
 * Sets the user's email address
 * @param user Pointer to user
 * @param email The email to set
 */
void user_set_email(User* user, const char* email) {
    if (user && email) {
        strncpy(user->email, email, sizeof(user->email) - 1);
        user->email[sizeof(user->email) - 1] = '\0';
    }
}

/**
 * Returns a formatted display string
 * @param user Pointer to user
 * @return Formatted display string (caller must free)
 */
char* user_display(const User* user) {
    if (!user) return NULL;
    
    char* buffer = (char*)malloc(128);
    if (!buffer) return NULL;
    
    snprintf(buffer, 128, "User %d: %s", user->id, user->name);
    return buffer;
}

/**
 * Frees a user structure
 * @param user Pointer to user to free
 */
void user_free(User* user) {
    if (user) {
        free(user);
    }
}

/**
 * User manager structure
 */
typedef struct {
    User** users;
    int count;
    int capacity;
} UserManager;

/**
 * Creates a new empty user manager
 * @return Pointer to new manager or NULL on failure
 */
UserManager* user_manager_new() {
    UserManager* manager = (UserManager*)malloc(sizeof(UserManager));
    if (!manager) return NULL;
    
    manager->users = NULL;
    manager->count = 0;
    manager->capacity = 0;
    
    return manager;
}

/**
 * Adds a user to the manager
 * @param manager Pointer to manager
 * @param user Pointer to user to add
 */
void user_manager_add(UserManager* manager, User* user) {
    if (!manager || !user) return;
    
    if (manager->count >= manager->capacity) {
        int new_capacity = manager->capacity == 0 ? 4 : manager->capacity * 2;
        User** new_users = (User**)realloc(manager->users, new_capacity * sizeof(User*));
        if (!new_users) return;
        
        manager->users = new_users;
        manager->capacity = new_capacity;
    }
    
    manager->users[manager->count++] = user;
}

/**
 * Retrieves a user by ID
 * @param manager Pointer to manager
 * @param id The user ID
 * @return Pointer to user or NULL if not found
 */
User* user_manager_get(UserManager* manager, int id) {
    if (!manager) return NULL;
    
    for (int i = 0; i < manager->count; i++) {
        if (manager->users[i]->id == id) {
            return manager->users[i];
        }
    }
    
    return NULL;
}

/**
 * Returns all users
 * @param manager Pointer to manager
 * @return Array of users (caller must not free)
 */
User** user_manager_list_all(UserManager* manager) {
    if (!manager) return NULL;
    return manager->users;
}

/**
 * Frees a user manager
 * @param manager Pointer to manager to free
 */
void user_manager_free(UserManager* manager) {
    if (manager) {
        if (manager->users) {
            free(manager->users);
        }
        free(manager);
    }
}

/**
 * Main entry point
 */
int main() {
    UserManager* manager = user_manager_new();
    
    User* user1 = user_new(1, "Alice");
    User* user2 = user_new(2, "Bob");
    
    user_manager_add(manager, user1);
    user_manager_add(manager, user2);
    
    User** users = user_manager_list_all(manager);
    for (int i = 0; i < manager->count; i++) {
        char* display = user_display(users[i]);
        if (display) {
            printf("%s\n", display);
            free(display);
        }
    }
    
    user_manager_free(manager);
    return 0;
}
